.github/workflows/cd.yml

```diff
@@ -27,25 +36,61 @@ jobs:
       uses: rlespinasse/github-slug-action@v4
 
     # Setup gcloud CLI
-    - name: Set up gcloud SDK environment
-      uses: google-github-actions/setup-gcloud@v0.4.0
+    - name: Authenticate to Google Cloud
+      id: auth
+      uses: google-github-actions/auth@v0.5.0
+      with:
+        credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+    # Setup Docker Buildx to allow use of docker cache layers from GH
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Login to Google Artifact Registry
+      uses: docker/login-action@v1.12.0
+      with:
+        registry: us-docker.pkg.dev
+        username: _json_key
+        password: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+    - name: Login to Google Container Registry
+      uses: docker/login-action@v1.12.0
+      with:
+        registry: gcr.io
+        username: _json_key
+        password: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+    # Build and push image to Google Artifact Registry
+    - name: Build & push
+      id: docker_build
+      uses: docker/build-push-action@v2.8.0
       with:
-        project_id: ${{ env.PROJECT_ID }}
-        service_account_key: ${{ secrets.GCLOUD_AUTH }}
-
-    # Build and push image to Google Container Registry
-    - name: Build
-      # Tagging w/ the commit SHA blocks the :latest tag on GCR
-      run: |
-        gcloud builds submit \
-          --config cloudbuild.yaml \
-          --substitutions SHORT_SHA="${{ env.GITHUB_SHA_SHORT }}",BRANCH_NAME="${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}"
-
-  deploy:
-    name: Deploy mainnet nodes
+        target: runtime
+        context: .
+        file: ./docker/Dockerfile.build
+        tags: |
+          ${{ env.GAR_BASE }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:latest
+          ${{ env.GAR_BASE }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:${{ env.GITHUB_SHA_SHORT }}
+          ${{ env.GCR_BASE }}/${{ env.GITHUB_REPOSITORY_SLUG_URL }}/${{ env.GITHUB_REPOSITORY_SLUG_URL }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:latest
+          ${{ env.GCR_BASE }}/${{ env.GITHUB_REPOSITORY_SLUG_URL }}/${{ env.GITHUB_REPOSITORY_SLUG_URL }}/${{ env.GITHUB_HEAD_REF_SLUG || env.GITHUB_REF_SLUG }}:${{ env.GITHUB_SHA_SHORT }}
+        build-args: |
+          NETWORK=${{ github.event.inputs.network || env.NETWORK }}
+          SHORT_SHA=${{ env.GITHUB_SHA_SHORT }}
+          RUST_BACKTRACE=1
+          ZEBRA_SKIP_IPV6_TESTS="1"
+          CHECKPOINT_SYNC=${{ github.event.inputs.checkpoint_sync || true }}
+          SENTRY_DSN=${{ secrets.SENTRY_ENDPOINT }}
+        push: true
+        cache-from: type=gha
+        cache-to: type=gha,mode=max
+
+  deploy-nodes:
+    name: Deploy Mainnet nodes
     needs: build
     runs-on: ubuntu-latest
     timeout-minutes: 30
+    if: github.event_name == 'push'
```

Comment:
I know that we only have `push` on `main` branch at the top, but would it hurt to also check the branch name in this `if`? For refactor-safety

Reply:
That might break manual `workflow_dispatch` checks on PR series?

It seems like an edge case, but it would be very confusing if it happened.

Reply:
🤔 The `workflow_dispatch` triggers the `deploy_instance` job below

Reply:
Fixed: https://github.com/ZcashFoundation/zebra/pull/3431/commits/22005a6ed18b90be52731ca4a439e3e3ccf0478a